equinox/_filters.py

```diff
@@ -135,21 +135,45 @@ def filter(
 def partition(
     pytree: PyTree,
     filter_spec: PyTree[AxisSpec],
+    *filter_specs: PyTree[AxisSpec],
```

Comment:
I dislike breaking backwards compatibility! Could you instead make ```filter_specs``` a keyword argument? This will be less ergonomic, since will force you to actually spell ```eqx.partition(..., filter_specs=())```, but it is worth it.

And, the _ordering_ of the filters matters in your implementation, so it is more 'logical' to have them as a sequence, and not as a variadic: in the current implementation it is not clear that order of filters matters a lot, but if it is another parameter, then it's pretty clear.

Reply:
FWIW if we made this a keyword then we'd need to allow not passing `filter_spec` itself in the case that it's used.

Hmm, this backward compatibility break may sink this whole endeavour.

Reply:
IMO forcing `replace` and `is_leaf` to be keyword arguments is actually an improvement as it could prevent user errors. It does break backward compatibility slightly, but I don't think there are many (if any) codebases using these arguments as positional. Note that `jax.tree_util.tree_map` forces `is_leaf` as a keyword argument.